soc/arm/st_stm32/common/Kconfig.soc

```diff
@@ -21,3 +21,8 @@ config USE_STM32_ASSERT
 	bool "STM32Cube HAL and LL drivers asserts"
 	help
 	  Enable asserts in STM32Cube HAL and LL drivers.
+
+config POWER_SUPPLY_SMPS
+	bool "Enable STM32 internal SMPS"
+	help
+	  Enable SMPS Power Supply.
```

Comment:
This config is currently supported only for STM32H7 series.
Can you encapsulate it with :
```
if SOC_SERIES_STM32H7X
...
endif 
```